Tag support for Drum of the Gathering

TheRealWormbo opened the pull request:
The drum now uses entity tags to determine which mobs can fill milk buckets and which shearable mobs can NOT be sheared: `botania:drum/milkable` (defaults to cow, mooshroom, and goat) and `botania:drum/no_shearing` (defaults to only mooshroom)
Additionally, if a brown mooshroom was fed a flower, the Drum of the Gathering will fill the first bowl for that mooshroom with the appropriate type of suspicious stew. Any additional bowls will be filled with mushroom stew as usual.
And finally, baby animals are excluded from filling buckets or bowls. (Whether they can be sheared is already determined by the mob's shearing logic, which usually excludes babies as well.)

(closes #4453)

TheRealWormbo committed:
Tag support for Drum of the Gathering (closing #4453)

The drum now uses entity tags to determine which mobs can fill milk buckets and which shearable mobs can NOT be sheared: `botania:drum/milkable` (defaults to cow, mooshroom, and goat) and `botania:drum/no_shearing` (defaults to only mooshroom)
Additionally, if a brown mooshroom was fed a flower, the Drum of the Gathering will fill the first bowl for that mooshroom with the appropriate type of suspicious stew. Any additional bowls will be filled with mushroom stew as usual.
And finally, baby animals are excluded from filling buckets or bowls. (Whether they can be sheared is already determined by the mob's shearing logic, which usually excludes babies as well.)

williewillus commented on Xplat/src/main/java/vazkii/botania/common/block/mana/DrumBlock.java:
```diff
@@ -64,26 +70,90 @@ public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, Co
+			SuspiciousStewItem.saveMobEffect(stewItem, effect, effectDuration);
+			spawnItem(mushroomCow, stewItem);
+
+			itemstack.shrink(1);
```
Use EntityHelper.shrinkItem which handles proper resyncing instead of calling bare shrink()

technically the callsite above should be using this helper too, but that loop always consumes the entire stack until it's gone so it's not strictly necessary

TheRealWormbo replied:
The other one probably wouldn't need to shrink the stack at all if it uses up the entire thing anyway.

williewillus commented on Xplat/src/main/java/vazkii/botania/common/block/mana/DrumBlock.java:
```diff
@@ -64,26 +70,90 @@ public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, Co
+		List<ItemEntity> items = world.getEntitiesOfClass(ItemEntity.class, mushroomCow.getBoundingBox(),
+				itemEntity -> itemEntity.getItem().is(Items.BOWL) && !itemEntity.getItem().isEmpty());
+		for (ItemEntity item : items) {
+			ItemStack itemstack = item.getItem();
```
let's call this something with `bowl` in it so it's clear which is which, it's easy to get mixed up when everything is `item` or `stack` or some combo

williewillus reviewed (commented):
couple small things

TheRealWormbo committed:
Improved variable/parameter names, added check for dead animals

Also added tests for dead animals and Bergamute interaction and removed redundant killAllEntities calls from test sequences
